rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py

```diff
@@ -183,12 +186,29 @@ def message(data):
             qos_profile=transient_qos,
         )
 
+        publisher_qos = QoSProfile(
+            history=History.KEEP_LAST,
+            depth=10,
+            reliability=Reliability.RELIABLE,
+            durability=Durability.VOLATILE)
+
         self.path_pub = self.create_publisher(
             PathRequest,
             'robot_path_requests',
             qos_profile=qos_profile_system_default,
         )
 
+        self.mode_pub = self.create_publisher(
+            ModeRequest,
+            'robot_mode_requests',
+            qos_profile=publisher_qos)
+
+        self.action_completed_pub = self.create_publisher(
+            ModeRequest,
+            'action_execution_notice',
+            qos_profile=qos_profile_system_default,
+        )
```

Comment:
Can't say I'm too familiar with `action_execution_notice`, but is it needed? What does it do in this case?

Reply:
`action_execution_notice` is published by the fleet manager to let the fleet adapter know that the robot has completed the perform action that it is carrying out, so that the fleet adapter can mark the action as finished.